### PR TITLE
Enabling Nullable Reference types in Calamari.Common and Shared

### PR DESCRIPTION
**Pull request by slewis74**
Included in here are code cleanup changes that we are rolling across the repos.

### CODE REVIEW

File: source/Calamari.Common/Features/Processes/Semaphores/LockFileBasedSemaphore.cs [hunk 1]
```diff
@@ -8,11 +8,11 @@ namespace Calamari.Common.Features.Processes.Semaphores
 {
     public class LockFileBasedSemaphore : ISemaphore
     {
-        public enum AquireLockAction
+        public enum AcquireLockAction
```
Comment (andrew-moore-octo) [inline source/Calamari.Common/Features/Processes/Semaphores/LockFileBasedSemaphore.cs:11]: I'm glad to see this fixed, but given the size of this PR, I would prefer to see things like this done separately.

File: source/Calamari.Common/Plumbing/Extensions/VariableExtensions.cs [hunk 2]
```diff
@@ -8,15 +8,12 @@ namespace Calamari.Common.Plumbing.Extensions
 {
     public static class VariableExtensions
     {
-        public static PathToPackage GetPathToPrimaryPackage(this IVariables variables, ICalamariFileSystem fileSystem, bool required)
+        public static PathToPackage GetPathToPrimaryPackage(this IVariables variables, ICalamariFileSystem fileSystem)
```
Comment (droyad) [inline source/Calamari.Common/Plumbing/Extensions/VariableExtensions.cs:11]: I added the `required` flag on purpose, likely for when converting the other command to use this method

File: source/Calamari.Common/Plumbing/IndentedTextWriter.cs [hunk 3]
```diff
@@ -476,13 +476,12 @@ class Indentation
     {
         readonly IndentedTextWriter writer;
         readonly int indent;
-        string s;
+        string? s;
 
         internal Indentation(IndentedTextWriter writer, int indent)
```
Comment (droyad) [inline source/Calamari.Common/Plumbing/IndentedTextWriter.cs:481]: `IndentedTextWriter` is not used, we can remove it.

File: source/Calamari.Shared/Integration/Certificates/CalamariCertificateStore.cs [hunk 4]
```diff
@@ -86,14 +86,15 @@ static X509Certificate2 GetOrAdd(string thumbprint, byte[] bytes, string passwor
...
         {
             return TryLoadCertificate(file, X509KeyStorageFlags.Exportable | X509KeyStorageFlags.MachineKeySet | X509KeyStorageFlags.PersistKeySet, true, password)
                 ?? TryLoadCertificate(file, X509KeyStorageFlags.Exportable | X509KeyStorageFlags.UserKeySet | X509KeyStorageFlags.PersistKeySet, true, password)
                 ?? TryLoadCertificate(file, X509KeyStorageFlags.Exportable | X509KeyStorageFlags.PersistKeySet, true, password)
                 ?? TryLoadCertificate(file, X509KeyStorageFlags.Exportable | X509KeyStorageFlags.MachineKeySet | X509KeyStorageFlags.PersistKeySet, false, password)
                 ?? TryLoadCertificate(file, X509KeyStorageFlags.Exportable | X509KeyStorageFlags.UserKeySet | X509KeyStorageFlags.PersistKeySet, false, password)
-                ?? TryLoadCertificate(file, X509KeyStorageFlags.Exportable | X509KeyStorageFlags.PersistKeySet, false, password);
+                ?? TryLoadCertificate(file, X509KeyStorageFlags.Exportable | X509KeyStorageFlags.PersistKeySet, false, password)
+                ?? throw new InvalidOperationException($"Unable to load certificate from file {file}");
```
Comment (droyad) [inline source/Calamari.Shared/Integration/Certificates/CalamariCertificateStore.cs:97]: nice

File: source/Calamari.Shared/Integration/Packages/Download/GitHubPackageDownloader.cs [hunk 5]
```diff
@@ -193,7 +192,10 @@ static bool TryParseVersion(string input, out IVersion version)
                         client.DownloadFileWithProgress(uri, tempPath, (progress, total) => log.Progress(progress, $"Downloading {packageId} v{version}"));
 
                         DeNestContents(tempPath, localDownloadName);
-                        return PackagePhysicalFileMetadata.Build(localDownloadName);
+                        var packagePhysicalFileMetadata = PackagePhysicalFileMetadata.Build(localDownloadName);
+                        if (packagePhysicalFileMetadata == null)
+                            throw new CommandException($"Unable to retrieve metadata for package {packageId}, version {version}");
```
Comment (droyad) [inline source/Calamari.Shared/Integration/Packages/Download/GitHubPackageDownloader.cs:197]: Would this be cleaner?
```suggestion
                        var packagePhysicalFileMetadata = PackagePhysicalFileMetadata.Build(localDownloadName)
                        ?? throw new CommandException($"Unable to retrieve metadata for package {packageId}, version {version}");
```

File: source/Calamari.Common/Features/Processes/Semaphores/FileLock.cs [hunk 6]
```diff
@@ -5,20 +5,31 @@
 
 namespace Calamari.Common.Features.Processes.Semaphores
 {
+    public interface IFileLock
+    {}
+    
     [DataContract]
-    public class FileLock
+    public class FileLock : IFileLock
```
[the remark this answers is not in the document]
Comment (slewis74) [reply source/Calamari.Common/Features/Processes/Semaphores/FileLock.cs:12]: 🤷

File: source/Calamari.Common/Features/Scripting/ScriptExecutor.cs [hunk 7]
```diff
@@ -14,18 +14,18 @@ public abstract class ScriptExecutor : IScriptExecutor
     {
         static readonly string CopyWorkingDirectoryVariable = "Octopus.Calamari.CopyWorkingDirectoryIncludingKeyTo";
 
-        public CommandResult Execute(Script script,
+        public CommandResult? Execute(Script script,
```
[the remark this answers is not in the document]
Comment (slewis74) [reply source/Calamari.Common/Features/Scripting/ScriptExecutor.cs:17]: This looks like it got changed due to something that got reverted along the way, but this didn't also get reverted.

File: source/Calamari.Common/Plumbing/IndentedTextWriter.cs (hunk 3 above)
Comment (slewis74) [reply source/Calamari.Common/Plumbing/IndentedTextWriter.cs:481]: ?, I tried, ConsoleLog is still using it.

**Pull request merged**